s3auth-relay/src/main/java/com/s3auth/relay/BaseFacade.java

```diff
@@ -0,0 +1,427 @@
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.s3auth.relay;
+
+import com.jcabi.aspects.Loggable;
+import com.jcabi.aspects.Tv;
+import com.jcabi.log.Logger;
+import com.jcabi.log.VerboseRunnable;
+import com.jcabi.log.VerboseThreads;
+import com.s3auth.hosts.Hosts;
+import java.io.Closeable;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.SynchronousQueue;
+import java.util.concurrent.TimeUnit;
+import javax.net.ssl.SSLServerSocketFactory;
+import javax.validation.constraints.NotNull;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.net.ftp.FTPReply;
+
+/**
+ * Convenient base superclass that creates a request dispatcher object for port
+ * listener implementations such as {@link HttpFacade} or {@link FtpFacade}.
+ *
+ * @author Simon Njenga (simtuje@gmail.com)
+ * @author Yegor Bugayenko (yegor@tpc2.com)
+ * @version $Id$
+ * @since 0.1
+ */
+@ToString
+@EqualsAndHashCode(
+    of = { "frontend", "backend", "sockets", "server", "secured" }
+)
+@SuppressWarnings({ "PMD.DoNotUseThreads", "PMD.TooManyMethods" })
+@Loggable(Loggable.DEBUG)
+class BaseFacade implements Closeable {
+
+    /**
+     * How many threads to use.
+     */
+    private static final int THREADS = Tv.HUNDRED;
+
+    /**
+     * Executor service, with socket openers.
+     */
+    private final transient ScheduledExecutorService frontend;
+
+    /**
+     * Executor service, with consuming threads.
+     */
+    private final transient ScheduledExecutorService backend;
+
+    /**
+     * Blocking queue of ready-to-be-processed sockets.
+     */
+    private final transient BlockingQueue<Socket> sockets =
+        new SynchronousQueue<Socket>();
+
+    /**
+     * Server socket.
+     */
+    private final transient ServerSocket server;
+
+    /**
+     * Secured Server socket.
+     */
+    private final transient ServerSocket secured;
+
+    /**
+     * Protected constructor.
+     * @param frontthreads The number of front threads
+     * @param frontprefix The name of the front threads
+     * @param backthreads The number of back threads
+     * @param backprefix The name of the back threads
+     * @param svr The server socket
+     * @param secure The secure server socket
+     * @checkstyle ParameterNumber (5 lines)
+     */
+    protected BaseFacade(final int frontthreads, final String frontprefix,
```

Comment:
@caarlos0 Once we create this object, it would mean we also have some ```getters``` in order to retrieve the number of threads and their name so as to pass them to this method ```Executors.newScheduledThreadPool(int corePoolSize, ThreadFactory threadFactory)```. We are restricted not to use ```getters```.